src/main/java/hudson/plugins/spotinst/jobs/SpotinstSyncGroupsOwner.java

```diff
@@ -0,0 +1,176 @@
+            LoggerFactory.getLogger(SpotinstSyncGroupsOwner.class);
+    private static       Set<GroupLockKey> groupsFromLastRun;
+    private static final Integer           lockTTLToSyncGroupRatio = 3;
+    public static final  Integer           JOB_INTERVAL_IN_SECONDS =
+            TimeHelper.getRedisTimeToLeaveInSeconds() / lockTTLToSyncGroupRatio;
+    final static         long              recurrencePeriod        = TimeUnit.SECONDS.toMillis(JOB_INTERVAL_IN_SECONDS);
+    //endregion
+
+    //region Constructor
+    public SpotinstSyncGroupsOwner() {
+        super("Sync Groups Owner");
+    }
+    //endregion
+
+    //region Public Methods
+    @Override
+    protected void execute(TaskListener taskListener) {
+        synchronized (this) {
+            List<BaseSpotinstCloud> activeClouds = getActiveClouds();
+            HandleCloudsChangeFromLastRun(activeClouds);
+
+            activeClouds.forEach(BaseSpotinstCloud::syncGroupOwner);
+        }
+    }
+
+    public void deallocateAll() {
+        List<Cloud>       cloudList = Jenkins.getInstance().clouds;
+        Set<GroupLockKey> groupLockAcquiringSet;
+
+        if (CollectionUtils.isNotEmpty(cloudList)) {
+            groupLockAcquiringSet = cloudList.stream().filter(cloud -> cloud instanceof BaseSpotinstCloud)
+                                             .map(cloud -> (BaseSpotinstCloud) cloud)
+                                             .map(spotinstCloud -> new GroupLockKey(spotinstCloud.getGroupId(),
+                                                                                    spotinstCloud.getAccountId()))
+                                             .collect(Collectors.toSet());
+        }
+        else {
+            groupLockAcquiringSet = new HashSet<>();
+        }
+
+        LOGGER.info(String.format("deallocating %s Spotinst clouds", groupLockAcquiringSet.size()));
+        deallocateGroups(groupLockAcquiringSet);
+        groupsFromLastRun = null;
+    }
+    //endregion
+
+    //region private methods
+    private List<BaseSpotinstCloud> getActiveClouds() {
+        List<Cloud>             clouds = Jenkins.getInstance().clouds;
+        List<BaseSpotinstCloud> retVal;
+
+        if (clouds != null) {
+            retVal = clouds.stream().filter(cloud -> cloud instanceof BaseSpotinstCloud)
+                           .map(cloud -> (BaseSpotinstCloud) cloud).filter(BaseSpotinstCloud::isActive)
+                           .collect(Collectors.toList());
+        }
+        else {
+            retVal = new ArrayList<>();
+        }
+
+        return retVal;
+    }
+
+    private void HandleCloudsChangeFromLastRun(List<BaseSpotinstCloud> activeClouds) {
+        Set<GroupLockKey> previousActiveGroups = getGroupsFromLastRun();
+        Set<GroupLockKey> currentActiveGroups  = getGroups(activeClouds);
+
+        previousActiveGroups.removeAll(currentActiveGroups);
+
+        deallocateGroups(previousActiveGroups);
```

Comment:
Add log: the groups %s are not in use anymoe by any active cloud, unlock them

Reply:
done